libraries/AP_InertialSensor/AP_InertialSensor.h

```diff
@@ -450,6 +418,39 @@ class AP_InertialSensor : AP_AccelCal_Client
     // force save of current calibration as valid
     void force_save_calibration(void);
 
+    // structure per harmonic notch filter. This is public to allow for
+    // easy iteration
+    class HarmonicNotch {
+    public:
+        HarmonicNotchFilterParams params;
+        HarmonicNotchFilterVector3f filter[INS_MAX_INSTANCES];
+
+        uint8_t num_dynamic_notches;
+
+        // the current center frequency for the notch
+        float calculated_notch_freq_hz[INS_MAX_NOTCHES];
+        uint8_t num_calculated_notch_frequencies;
+
+        // Update the harmonic notch frequency
+        void update_notch_freq_hz(float scaled_freq);
+
+        // Update the harmonic notch frequencies
+        void update_notch_frequencies_hz(uint8_t num_freqs, const float scaled_freq[]);
+
+        // runtime update of notch parameters
+        void update_params(uint8_t instance, bool converging, float gyro_rate);
+
+        // Update the harmonic notch frequencies
+        void update_freq_hz(float scaled_freq);
+        void update_frequencies_hz(uint8_t num_freqs, const float scaled_freq[]);
+        
+    private:
+        // support for updating harmonic filter at runtime
+        float last_center_freq_hz;
+        float last_bandwidth_hz;
+        float last_attenuation_dB;
+    } harmonic_notches[HAL_INS_NUM_HARMONIC_NOTCH_FILTERS];
```

Comment:
I thought we were going with the convention of preceeding _ for members in this file

Reply:
I've never liked that convention actually, and even when the convention is used it is only for public members